#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_new_style_classes.dpatch by Thomas Viehmann <tv@beamnet.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix SOAPBuilder.py dump to gracefully handle new style classes
## DP: such as faulType (new style in python2.5 as subclass of
## DP: Exception)

@DPATCH@
--- trunk~/SOAPpy/SOAPBuilder.py	2005-02-21 21:24:13.000000000 +0100
+++ trunk/SOAPpy/SOAPBuilder.py	2008-11-29 16:57:21.000000000 +0100
@@ -281,12 +281,14 @@
         if Config.debug: print "In dump.", "obj=", obj
         ns_map = ns_map.copy()
         self.depth += 1
-
         if type(tag) not in (NoneType, StringType, UnicodeType):
             raise KeyError, "tag must be a string or None"
-
+        if isinstance(obj, anyType):
+            methname = "dump_instance"
+        else:
+            methname = "dump_" + type(obj).__name__
         try:
-            meth = getattr(self, "dump_" + type(obj).__name__)
+            meth = getattr(self, methname)
         except AttributeError:
             if type(obj) == LongType:
                 obj_type = "integer"
@@ -392,7 +394,7 @@
         tag = tag or self.gentag()
         tag = toXMLname(tag) # convert from SOAP 1.2 XML name encoding
 
-        if type(obj) == InstanceType:
+        if type(obj) == InstanceType or isinstance(obj, anyType):
             data = obj.data
         else:
             data = obj
